src/services/contactService.ts

```diff
@@ -1,53 +1,153 @@
     .then((contact: Party) => {
-      debug(`createContact(${JSON.stringify(args)}) succeeded`);
+      logger.logEvent({
+        type: LoggingEventType.AUDIT,
+        data: {
+          description: 'contact created',
+          actionType: ActionType.CREATE,
+          actionSubType: 'create contact',
+          initiatorType: InitiatorType.USER,
+          data: contact,
+          diagnosticData: args,
+        },
+      });
+
       return contact;
     })
     .catch((error: Error) => Promise.reject(Error(`Unable to create contact. Error: ${error}`)));
 };
 
 export const updateContact = async (args: IUpdateContactArgs): Promise<Party> => {
-  debug(`updateContact(${JSON.stringify(args)})...`);
+  await logger.logEvent({
+    type: LoggingEventType.AUDIT,
+    data: {
+      level: LogLevel.TRACE,
+      description: 'updateContact function call',
+      actionType: ActionType.UPDATE,
+      actionSubType: 'update contact',
+      initiatorType: InitiatorType.USER,
+      diagnosticData: args,
+    },
+  });
+
   return cmUpdateContact(args)
     .then((contact: Party) => {
-      debug(`updateContact(${JSON.stringify(args)}) succeeded`);
+      logger.logEvent({
+        type: LoggingEventType.AUDIT,
+        data: {
+          description: 'contact updated',
+          actionType: ActionType.UPDATE,
+          actionSubType: 'update contact',
+          initiatorType: InitiatorType.USER,
+          data: contact,
+          diagnosticData: args,
+        },
+      });
+
       return contact;
     })
     .catch((error: Error) => Promise.reject(Error(`Unable to update contact. Error: ${error}`)));
 };
 
 export const removeContact = async (args: RemoveContactArgs): Promise<boolean> => {
-  debug(`removeContact(${JSON.stringify(args)})...`);
+  await logger.logEvent({
+    type: LoggingEventType.AUDIT,
+    data: {
+      level: LogLevel.TRACE,
+      description: 'removeContact function call',
+      actionType: ActionType.DELETE,
+      actionSubType: 'remove contact',
+      initiatorType: InitiatorType.USER,
+      diagnosticData: args,
+    },
+  });
+
   return cmRemoveContact(args)
     .then((isDeleted: boolean) => {
-      debug(`removeContact(${JSON.stringify(args)}) succeeded`);
+      logger.logEvent({
+        type: LoggingEventType.AUDIT,
+        data: {
+          description: isDeleted ? 'contact removed' : 'contact not removed',
+          actionType: ActionType.DELETE,
+          actionSubType: 'remove contact',
+          initiatorType: InitiatorType.USER,
+          data: isDeleted,
+          diagnosticData: args,
+        },
+      });
+
       return isDeleted;
     })
     .catch((error: Error) => Promise.reject(Error(`Unable to remove contact. Error: ${error}`)));
 };
 
 export const addIdentity = async (args: IAddIdentityArgs): Promise<Identity> => {
-  debug(`addIdentity(${JSON.stringify(args)})...`);
+  await logger.logEvent({
+    type: LoggingEventType.AUDIT,
+    data: {
+      level: LogLevel.TRACE,
```

Comment:
Not trace level; This one will also have a similar issue as the getContacts. Can be user initiated in the future, but also system initiated

Reply:
removed trace. and we do not have a way to see the difference